x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/FileOperatorUsersStore.java

```diff
@@ -129,49 +143,81 @@ static final class Group {
         private static final Set<String> SINGLETON_REALM_TYPES = Set.of(
             FileRealmSettings.TYPE,
             NativeRealmSettings.TYPE,
-            ReservedRealm.TYPE
+            ReservedRealm.TYPE,
+            ServiceAccountSettings.REALM_TYPE
         );
 
         private final Set<String> usernames;
         private final String realmName;
         private final String realmType;
+        private final String tokenSource;
+        private final Set<String> tokenNames;
         private final Authentication.AuthenticationType authenticationType;
 
         Group(Set<String> usernames) {
             this(usernames, null);
         }
 
         Group(Set<String> usernames, @Nullable String realmName) {
-            this(usernames, realmName, null, null);
+            this(usernames, realmName, null, null, null, null);
         }
 
-        Group(Set<String> usernames, @Nullable String realmName, @Nullable String realmType, @Nullable String authenticationType) {
+        Group(
+            Set<String> usernames,
+            @Nullable String realmName,
+            @Nullable String realmType,
+            @Nullable String authenticationType,
+            @Nullable String tokenSource,
+            @Nullable Set<String> tokenNames
```

Comment:
Since one of the main value propositions for service accounts is multiple tokens for the same service account I believe that token_names needs to be a collection. Multiple names are straight forward for file based accounts since the configuration would know the names up front and for index based you could preemptively add additional names if you think the service will need to scale (or we can add in wildcard support in the future). 

> And we will take elastic/kibana and fleet-token as a valid combination, which is likely wrong. 

I think this would be just a misconfiguration and isn't that harmful beyond that it's confusing.

Reply:
If this is a misconfiguration, I think we should validate against it, i.e. prevent multiple usernames + multiple tokenNames. This ambiguity bothers me.

Reply:
I meant a misconfiguration in the sense of a typo is a misconfiguration. We can protect against all misconfigurations. IMO the value of having multiple tokens names > the possibility of misconfiguration.  This is internal only configuration so we can advise to not configure it a way that is ambiguous or confusing.

Reply:
I still have a different view. But overall it's probably not worth holding this pr.

Reply:
> prevent multiple usernames + multiple tokenNames. This ambiguity bothers me.

Oh ... I missed the main point. I can add some validation against having multiple usernames when auth_type is token.  (in a follow up preferably)